third_party/airflow/armada/operators/armada.py

```diff
@@ -131,3 +131,37 @@ def _get_lookout_url(self, job_id: str) -> str:
         if self.lookout_url_template is None:
             return ""
         return self.lookout_url_template.replace("<job_id>", job_id)
+
+
+
+    def on_kill(self) -> None:
+            """
+            Stops the JobService from listening to the JobSet.
+            return None
+            """
+            if self.job_id is not None:
+                try:
+                    self.job_service.unsubscribe(self.queue, self.job_set_id, self.job_id)
+                except Exception as e:
+                    armada_logger.warning("Error unsubscribing from JobSet: %s", str(e))
+
```

Comment:
I don't think this code would work at all.  

In on_kill, you want to call `self.armada_client.cancel_jobset(self.queue, self.job_set_id)` but you never save `self.queue` and `self.job_set_id` in the class.  

So:

1) I'd remove the cancel_jobs function.  You don't want that and you don't want interactive ways of cancelling jobs.  
2) Please add the cancelling of jobs in the on_kill function by passing a job_set and queue.

Reply:
ok

Reply:
Any update on this?

Reply:
I am sorry @kannon92 , actually from past few I am busy in my university exam and writing cover letter. I will soon back on work.